src/core/server/security/types.ts

```diff
@@ -20,3 +21,21 @@ export interface IReadOnlyService {
     hideCapabilities: Partial<Capabilities>
   ): Promise<Partial<Capabilities>>;
 }
+
+export interface IdentityEntry {
+  name: string;
+  id: string;
+}
+
+export interface IdentitySourceHandler {
```

Comment:
```suggestion
/**
 *    The identitySource handler primarily facilitates role-based authentication as the OpenSearch Security plugin employs this approach.
 *    In role-based authentication, there are two main concepts: user/role, and the handler is designed to primarily handle these terms.
 **/
export interface IdentitySourceHandler {
```
Nit: some comments may help to illustrate why we design these interfaces.

Reply:
Thanks, I have updated the comments